pkgs/by-name/bl/bleep/package.nix

```diff
@@ -0,0 +1,91 @@
+{
+  stdenv,
+  stdenvNoCC,
+  fetchurl,
+  autoPatchelfHook,
+  installShellFiles,
+  makeWrapper,
+  lib,
+  zlib,
+}:
+let
+  platform =
+    {
+      x86_64-linux = "x86_64-pc-linux";
+      x86_64-darwin = "x86_64-apple-darwin";
+      aarch64-darwin = "arm64-apple-darwin";
+    }
+    ."${stdenv.system}";
+
+  hash =
+    {
+      x86_64-linux = "sha256-ttfmTMSK6u2zX87CUYjibr675Xzn4/byWub67e5Ir+I=";
+      x86_64-darwin = "sha256-bh1yQ7KZ+cwVHTLG+gpYDLPKG2VUcrz88Sqd2tRiKfM=";
+      aarch64-darwin = "sha256-IV2JPXibv80Odb+1Zkw1YUoutHsi5YaIjyNpAOIUlr4=";
+    }
+    ."${stdenv.system}";
+  version = "0.0.7";
+in
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "bleep";
+  version = version;
+
+  src = fetchurl {
+    url = "https://github.com/oyvindberg/bleep/releases/download/v${version}/bleep-${platform}.tar.gz";
+    sha256 = hash;
+  };
+
+  nativeBuildInputs = [
+    installShellFiles
+    makeWrapper
+  ] ++ lib.optional stdenv.isLinux autoPatchelfHook;
+
+  buildInputs = [
+    zlib
+    stdenv.cc.cc
+  ];
+
+  unpackPhase = ''
+    runHook preUnpack
+    tar -xf $src
+    runHook postUnpack
+  '';
```

Comment:
Forgive me if I'm wrong, but isn't this `unpackPhase` redundant? I thought `.tar.gz` were automatically unpacked.

Reply:
I think that is true if you use [`fetchzip`](https://nixos.org/manual/nixpkgs/stable/#sec-pkgs-fetchers-fetchzip) (which seems like the most appropriate fetcher to use).

Reply:
`fetchzip` does unpack as part of its own thing.

`mkDerivation`s default `unpackPhase` does unpack `$src` if it is a tarball or zip (IIRC for ZIP you need to have zip in the `nativeBuildInputs`) or it copies recursively if `$src` is plain folder.

Reply:
Oh you are right. Which of the two would be the preferred approach (if there is one)?